bosh-stemcell/spec/support/stemcell_opensuse_shared_examples.rb

```diff
@@ -0,0 +1,134 @@
+      end
+
+      it('should be of mode 600 (stig: V-38583)') { should be_mode(0600) }
+      it('should be owned by root (stig: V-38579)') { should be_owned_by('root') }
+      it('should be grouped into root (stig: V-38581)') do
+        expect(subject.group).to eq('root')
+      end
+    end
+
+    # GRUB 0.97 configuration (used only on Amazon PV hosts) must have same kernel params as GRUB 2
+    describe file('/boot/grub/grub.conf') do
+      its(:content) { should match 'net.ifnames=0' }
+      its(:content) { should match 'selinux=0' }
+      its(:content) { should match 'plymouth.enable=0' }
+      its(:content) { should_not match 'xen_blkfront.sda_is_xvda=1' }
+
+      it('should be of mode 600 (stig: V-38583)') { should be_mode(0600) }
+      it('should be owned by root (stig: V-38579)') { should be_owned_by('root') }
+      it('should be grouped into root (stig: V-38581)') do
+        expect(subject.group).to eq('root')
+      end
+      it('audits processes that start prior to auditd (CIS-8.1.3)') do
+        expect(subject.content).to match ' audit=1'
+      end
+    end
+  end
+
+  context 'installed by bosh_harden' do
+    describe 'disallow unsafe setuid binaries' do
+      subject { command('find / -xdev -perm /6000 -a -type f').stdout.split }
+
+      it { should match_array(%w(/usr/bin/su /usr/bin/sudo)) }
+    end
+
+    describe 'disallow root login' do
+      subject { file('/etc/ssh/sshd_config') }
+
+      its(:content) { should match /^PermitRootLogin no$/ }
+    end
+  end
+
+  context 'installed by system-network on all IaaSes', { exclude_on_warden: true } do
+    describe file('/etc/hostname') do
+      it { should be_file }
+      its (:content) { should eq('bosh-stemcell') }
+    end
+  end
+
+  context 'installed by bosh_aws_agent_settings', {
+    exclude_on_openstack: true,
+    exclude_on_vcloud: true,
+    exclude_on_vsphere: true,
+    exclude_on_warden: true,
+    exclude_on_azure: true,
+  } do
+    describe file('/var/vcap/bosh/agent.json') do
+      it { should be_valid_json_file }
+      its(:content) { should match('"Type": "HTTP"') }
+    end
+  end
+
+  context 'installed by bosh_google_agent_settings', {
+    exclude_on_aws: true,
+    exclude_on_openstack: true,
+    exclude_on_vcloud: true,
+    exclude_on_vsphere: true,
+    exclude_on_warden: true,
+    exclude_on_azure: true,
+  } do
+    describe file('/var/vcap/bosh/agent.json') do
+      it { should be_valid_json_file }
+      its(:content) { should match '"Type": "InstanceMetadata"' }
+    end
+  end
+
+  context 'installed by bosh_vsphere_agent_settings', {
+    exclude_on_aws: true,
+    exclude_on_vcloud: true,
+    exclude_on_openstack: true,
+    exclude_on_warden: true,
+    exclude_on_azure: true,
+   } do
+    describe file('/var/vcap/bosh/agent.json') do
+      it { should be_valid_json_file }
+      its(:content) { should match('"Type": "CDROM"') }
+    end
+  end
+
+  context 'installed by bosh_azure_agent_settings', {
```

Comment:
Should have tests for `bosh_softlayer_agent_settings` and `bosh_openstack_agent_settings` as well.